### PR TITLE
Python 3: port xhr tests [part 6]

### PR DESCRIPTION
**Pull request by ziransun**

### CODE REVIEW

File: xhr/resources/infinite-redirects.py [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
-    new_location = "%s?page=%s&type=%s&mix=%s" % (location, page, type, mix)
-    headers = [("Cache-Control", "no-cache"),
-               ("Pragma", "no-cache"),
-               ("Location", new_location)]
-    return 301, headers, "Hello guest. You have been redirected to " + new_location
+    new_location = u"%s?page=%s&type=%s&mix=%s" % (location, page, type, mix)
+    headers = [(b"Cache-Control", b"no-cache"),
+               (b"Pragma", b"no-cache"),
+               (b"Location", isomorphic_encode(new_location))]
+    return 301, headers, b"Hello guest. You have been redirected to " + isomorphic_encode(new_location)
```
Comment (stephenmcgruer) [inline xhr/resources/infinite-redirects.py:24]: Can we keep in text and avoid the `isomorphic_encode` here? E.g.

```suggestion
    return 301, headers, u"Hello guest. You have been redirected to " + new_location
```

File: xhr/resources/parse-headers.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 def main(request, response):
     content = ""
```
Comment (stephenmcgruer) [inline xhr/resources/parse-headers.py:2]:
```suggestion
    content = u""
```

File: xhr/resources/infinite-redirects.py (hunk 1 above)
Comment (ziransun) [reply xhr/resources/infinite-redirects.py]: Code updated. Thank you!

File: xhr/resources/parse-headers.py (hunk 2 above)
Comment (ziransun) [reply xhr/resources/parse-headers.py]: Added.

**Review by stephenmcgruer — APPROVED** [review]

**Pull request merged**